lockfile/src/parsers/yarn.rs

```diff
@@ -29,28 +35,80 @@ fn entry(input: &str) -> IResult<&str, Package> {
     Ok((i, my_entry))
 }
 
-fn parse_entry(input: &str) -> IResult<&str, Package> {
-    context("entry", tuple((entry_name, entry_version)))(input).map(|(next_input, res)| {
-        let (name, version) = res;
-        (next_input, Package {
-            name: name.to_string(),
-            version: PackageVersion::FirstParty(version.to_string()),
-            package_type: PackageType::Npm,
-        })
-    })
+fn parse_entry(input: &str) -> IResult<&str, Option<Package>> {
+    let (input, (name, version)) = context("entry", tuple((entry_name, entry_version)))(input)?;
+
+    let version = match version {
+        Some(version) => version,
+        None => return Ok((input, None)),
+    };
+
+    let package = Package { version, name: name.to_string(), package_type: PackageType::Npm };
+
+    Ok((input, Some(package)))
 }
 
 fn entry_name(input: &str) -> IResult<&str, &str> {
+    // Strip optional quotes.
     let (i, _) = opt(tag(r#"""#))(input)?;
+
+    // Strip optional aliased package name.
+    let (i, _) = recognize(opt(tuple((take_until("@npm:"), tag("@npm:")))))(i)?;
+
+    // Allow for up to one leading `@` in package name (like `@angular/cli`).
     let opt_at = opt(tag("@"));
-    let name = tuple((opt_at, take_until("@")));
-    context("name", recognize(name))(i)
+
+    // Consume everything until version separator as name.
+    let name_parser = tuple((opt_at, take_until("@")));
+    context("name", recognize(name_parser))(i)
 }
 
-fn entry_version(input: &str) -> IResult<&str, &str> {
+fn entry_version(input: &str) -> IResult<&str, Option<PackageVersion>> {
+    // Handle path dependencies.
+    if input.starts_with("@./") || input.starts_with("@../") || input.starts_with("@/") {
+        return path_dep(input);
+    }
+
+    // Handle git dependencies.
+    if input.starts_with("@git://") {
+        return git_dep(input);
+    }
+
+    // Ignore HTTP(S) dependencies.
+    //
+    // These could be either git or tar dependencies, so to avoid miscategorization
+    // we just ignore them.
+    if input.starts_with("@http://") || input.starts_with("@https://") {
+        return Ok((input, None));
+    }
```

Comment:
Could you fill me in on the reasoning for ignoring these dependencies or link to any discussion threads about it? I know it says to avoid miscategorization between the two options, but I'm trying to understand the bigger picture of why this problem means we have to completely leave them out of analysis.

Reply:
I mean the comment says it all. We don't know if it's a link to a tar package or a git dependency. So if we don't know, we should omit it rather than reporting it incorrectly.

Reply:
I think I answered my own question. Looks like we submit packages as triples (not purls) and we [don't submit HTTP packages at all anyways](https://github.com/phylum-dev/cli/blob/main/cli/src/commands/parse.rs#L363). I figured we'd have some ability to process packages at arbitrary URLs / git repos, but I guess we don't.